src/components/util/calendar.js

```diff
@@ -0,0 +1,61 @@
+import dayjs from "dayjs";
+
+export const generateDate = (
+    month = dayjs().month(),
+    year = dayjs().year()
+) => {
+    const firstDateOfMonth = dayjs().year(year).month(month).startOf("month");
+    const lastDateOfMonth = dayjs().year(year).month(month).endOf("month");
+
+    const arrayOfDate = [];
+
+    // create prefix date
+    for (let i = 0; i < firstDateOfMonth.day(); i++) {
+        const date = firstDateOfMonth.day(i);
+
+        arrayOfDate.push({
+            currentMonth: false,
+            date,
+        });
+    }
+
+    // generate current date
+    for (let i = firstDateOfMonth.date(); i <= lastDateOfMonth.date(); i++) {
+        arrayOfDate.push({
+            currentMonth: true,
+            date: firstDateOfMonth.date(i),
+            today:
+                firstDateOfMonth.date(i).toDate().toDateString() ===
+                dayjs().toDate().toDateString(),
+        });
+    }
+
+    const remaining = 42 - arrayOfDate.length;
```

Comment:
not sure about what's this the meaning of this line

Reply:
this has been explained

Reply:
this line is for generate sufix for the calendar in other wrds this generates the days that are not included at this month but in the next month